synapse_token_authenticator/token_authenticator.py

```diff
@@ -67,20 +75,15 @@ def __init__(self, config: dict, account_handler: ModuleApi):
                 self.LoginMetadataResource(oidc),
             )
 
-        if (custom_flow := getattr(self.config, "custom_flow", None)) is not None:
-            if custom_flow.secret:
-                k = {
-                    "k": base64.urlsafe_b64encode(
-                        custom_flow.secret.encode("utf-8")
-                    ).decode("utf-8"),
-                    "kty": "oct",
-                }
-                self.custom_flow_key = jwk.JWK(**k)
-            else:
-                with open(custom_flow.keyfile) as f:
-                    self.key = jwk.JWK.from_pem(f.read())
-            auth_checkers[("com.famedly.login.token.custom", ("token",))] = (
-                self.check_custom_flow
+        if (cfg := getattr(self.config, "oauth", None)) is not None:
+            if cfg.expose_metadata_resource:
+                self.api.register_web_resource(
+                    f"/_famedly/login/{cfg.expose_metadata_resource.name}",
```

Comment:
Am I using the `expose_metadata_resource` property in the `homeserver.yaml` configuration correctly:

```homeserver.yaml
oauth:
  expose_metadata_resource:
    name: test
    issuer: http://localhost:8080/realms/matrix
```

If yes, the type of `expose_metadata_resource` is still dict from the yaml parsing. Therefore, I get the error is `AttributeError: 'dict' object has no attribute 'name'` when I try to use the property.
If the property stays a `dict` we should access its element like this:
```suggestion
                    f"/_famedly/login/{cfg.expose_metadata_resource['name']}",
```

Reply:
fixed